pkgs/development/tools/reshape/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, fetchpatch
+, git
+, nixosTests
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "reshape";
+  version = "0.6.1";
+
+  src = fetchFromGitHub {
+    owner = "fabianlindfors";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-iX8qhDk0PP0AB3bJ6ck8AZ9SCErFH416ggAkgK8O900=";
+  };
+
+  cargoPatches = [
+    ./add-Cargo.lock.patch
+  ];
+
+  cargoHash = "sha256-UL/vP8055JRbbf5bqf0V3wGh+iF9ztVhyoMrtNs/c+4=";
+
+  # Disabled because requires a running Postgres instance.
+  doCheck = false;
```

Comment:
You can use `postgresqlTestHook` to spin up a temporary postgresql instance for testing.

Reply:
I couldn't find a way to create new env variables between postgres setting `PGHOST` and the tests executing.

This is required because the tests read the `POSTGRES_CONNECTION_STRING` which is not set by postgres.

Reply:
Is `preCheck` executed in the right time window?

Reply:
It seems to be executed before Postgres

Reply:
Got a very strange idea:
```
postBuild = ''
 preCheckHooks+=('postgresqlConn')
 postgresqlConn() {
     export POSTGRES_CONNECTION_STRING="$PGHOST"
 }
''
```
But we really should have a way of running custom commands after `postgresqlTestHook` and before the checks run.

Reply:
You can now set the connection string in `postgresqlTestSetupPost` (need to do a rebase first)
See https://github.com/NixOS/nixpkgs/pull/212555